receptorctl/receptorctl/socket_interface.py

```diff
@@ -33,27 +34,49 @@ def read_and_parse_json(self):
         data = json.loads(text)
         return data
 
+    def readyaml(self, ctxobj):
+        yamlfile = ctxobj["yaml"]
+        tlsclient = ctxobj["tlsclient"]
+        if yamlfile and tlsclient:
+            with open(yamlfile, "r") as yam:
+                self.yaml = yaml.load(yam, Loader=yaml.FullLoader)
+                yam.close()
+            for i in self.yaml:
+                key = i.get("tls-client", None)
+                if key:
+                     if key["name"] == tlsclient:
+                         ctxobj["key"] = key.get("key", ctxobj["key"]) # if not in yaml, keep the previous value
+                         ctxobj["cert"]= key.get("cert", ctxobj["cert"])
+                         ctxobj["rootcas"] = key.get("rootcas", ctxobj["rootcas"])
+                         ctxobj["insecureskipverify"] = key.get("insecureskipverify", ctxobj["insecureskipverify"])
+                         break
+
     def simple_command(self, command):
         self.writestr(f"{command}\n")
         return self.read_and_parse_json()
 
-    def connect(self, address):
+    def connect(self, ctxobj):
+        address = ctxobj["socket"]
+        key = ctxobj["key"]
+        cert = ctxobj["cert"]
+        rootcas = ctxobj["rootcas"]
+        insecureskipverify = ctxobj["insecureskipverify"]
         if self.socket is not None:
             raise ValueError("Already connected")
-        m = re.compile("tcp:(//)?([a-zA-Z0-9-]+):([0-9]+)|(unix:(//)?)?([^:]+)").fullmatch(address)
+        m = re.compile("(tcp|tls):(//)?([a-zA-Z0-9-]+):([0-9]+)|(unix:(//)?)?([^:]+)").fullmatch(address)
         if m:
-            if m[6]:
-                path = os.path.expanduser(m[6])
+            if m[7]:
```

Comment:
This isn't great code style (I can say that since it was me who originally wrote it).  It would be better to assign the `m[x]` values to named variables up top, and then refer to them by name in later code.

Reply:
fixed